health-monitor/src/index.ts

```diff
@@ -0,0 +1,185 @@
+
+export interface HealthCheckResult {
+	authServerStatus: number;
+	chainDbWatcherServerStatus: number;
+	hasuraServerStatus: number;
+	latestBlockNumber: number;
+	onchainLinkReferendumDelays: boolean;
+	postsWithAuthor: boolean;
+	prismaVersion: string;
+	reactServerStatus: number;
+	referendumDelays: boolean;
+}
+
+async function getLatestBlockNumber (): Promise<number> {
+	const provider = new WsProvider(process.env.ARCHIVE_NODE_ENDPOINT);
+	const api = await ApiPromise.create({ provider });
+
+	const latestBlockNumber = (
+		await api.derive.chain.bestNumber()
+	).toNumber();
+
+	return latestBlockNumber;
+}
+
+async function getPostsWithAuthor (): Promise<boolean> {
+	return fetch(`${process.env.HASURA_SERVER}/v1/graphql`, {
+		body: JSON.stringify({
+			operationName: null,
+			query: `query LatestDiscussionPosts($limit: Int!) {
+				posts(order_by: {created_at: desc}, limit: $limit) {
+				  id
+				  author {
+					id
+					name
+					username
+				  }
+				}
+			}`,
+			variables: { limit: 10 }
+		}),
+		headers: {
+			'content-type': 'application/json'
+		},
+		method: 'POST'
+	}).then(res => res.json()).then(({ data }) => {
+		const result = data?.posts?.reduce((acc: boolean, post: any) => {
+			return acc && !!post?.id && !!post?.author?.id;
+		}, true);
+
+		return result && data?.posts?.length > 0;
+	});
+}
+
+async function getPrismaVersion (): Promise<string> {
+	return fetch(`${process.env.CHAIN_DB_SERVER}/management`, {
+		body: JSON.stringify({
+			operationName: null,
+			query: `{
+				serverInfo {
+					version
+				}
+			}`,
+			variables: {}
+		}),
+		headers: {
+			'content-type': 'application/json'
+		},
+		method: 'POST'
+	}).then(res => res.json()).then(({ data }) => data?.serverInfo?.version);
+}
+
+async function getReferendumDelays (): Promise<boolean> {
+	return fetch(`${process.env.CHAIN_DB_SERVER}`, {
+		body: JSON.stringify({
+			operationName: null,
+			query: `{
+				referendums(last:10){
+				  delay
+				}
+			}`,
+			variables: {}
+		}),
+		headers: {
+			'content-type': 'application/json'
+		},
+		method: 'POST'
+	}).then(res => res.json()).then(({ data }) => data?.referendums?.length > 0);
+}
+
+async function getOnchainLinkReferendumDelays (): Promise<boolean> {
+	return fetch(`${process.env.HASURA_SERVER}/v1/graphql`, {
+		body: JSON.stringify({
+			operationName: null,
+			query: `query LatestDemocracyTreasuryProposalPosts($limit: Int!) {
+				posts(limit: $limit, where: {onchain_link: {onchain_referendum_id: {_is_null: false}}}) {
+				  	id
+				  	onchain_link {
+						id
+						onchain_referendum {
+					  		delay
+						}
+				  	}
+				}
+			}`,
+			variables: { limit: 10 }
+		}),
+		headers: {
+			'content-type': 'application/json'
+		},
+		method: 'POST'
+	}).then(res => res.json()).then(({ data }) => {
+		const result = data?.posts?.reduce((acc: boolean, post: any) => {
+			return acc && !!post?.id && !!post?.onchain_link?.onchain_referendum?.delay;
```

Comment:
I might have given a bad example, I think delay can be `0` sometimes. Let's rather take the `voteThreshold`.
```suggestion
			return acc && !!post?.id && !!post?.onchain_link?.onchain_referendum?.voteThreshold;
```